linode/instance/resource.go

```diff
@@ -53,50 +53,30 @@ func readResource(ctx context.Context, d *schema.ResourceData, meta interface{})
 		return diag.Errorf("Error parsing Linode instance ID %s as int: %s", d.Id(), err)
 	}
 
-	var instance *linodego.Instance
-	var instanceNetwork *linodego.InstanceIPAddressResponse
-	var instanceDisks []linodego.InstanceDisk
-	var instanceConfigs []linodego.InstanceConfig
-
-	err = helper.RunBatch(ctx,
-		func(ctx context.Context) (err error) {
-			instance, err = client.GetInstance(ctx, id)
-			if err != nil {
-				err = fmt.Errorf("failed to get instance: %w", err)
-			}
-			return
-		},
-		func(ctx context.Context) (err error) {
-			instanceNetwork, err = client.GetInstanceIPAddresses(ctx, id)
-			if err != nil {
-				err = fmt.Errorf("failed to get instance networking: %w", err)
-			}
-			return
-		},
-		func(ctx context.Context) (err error) {
-			instanceDisks, err = client.ListInstanceDisks(ctx, id, nil)
-			if err != nil {
-				err = fmt.Errorf("failed to get instance disks: %w", err)
-			}
-			return
-		},
-		func(ctx context.Context) (err error) {
-			instanceConfigs, err = client.ListInstanceConfigs(ctx, id, nil)
-			if err != nil {
-				err = fmt.Errorf("failed to get instance configs: %w", err)
-			}
-			return
-		},
-	)
+	instance, err := client.GetInstance(ctx, id)
+	if linodego.IsNotFound(err) {
```

Comment:
linodego.IsNotFound(...) has handling for nil errors so I figured I'd leave it out of the `if err != nil` block for readability 👍